CONTRIBUTING.md

```diff
@@ -1,44 +1,216 @@
 
-## 6. Test your code
+##### [Additional Notes](#additional-notes)
 
-Please, follow test policy for the code you write. New code should be covered by at least 80% (see coverage reports in pull requests). 
+- [Informational Labels](#informational-labels)
+- [Pull Request and Issue Labels](#pull-request-and-issue-labels)
+- [Issue Labels](#issue-labels)
+- [Pull Request Labels](#pull-request-labels)
+- [Contact Developers](#contact-developers)
 
-## 7. Create pull request
 
-Follow gitflow approach to create a branch for your code:
-* feature/whatever-feature-you-implement, create a branch from develop
-* fix/whatever-you-fix-in-develop
-* hotfix/whatever-you-fix-in-master
 
-Follow the flow in Iroha working agreement.
+## How Can I Contribute?
 
-Thank you for your interest to Iroha project!
+### Reporting bugs
+
+*Bug* is an error, design flaw, failure or fault in Iroha that causes it to produce an incorrect or unexpected result, or to behave in unintended ways. 
+
+Bugs are tracked as [GitHub Issues](https://guides.github.com/features/issues/). To submit a bug, create new Issue and include these details:
+- **Title**
+    - Write prefix `[Bug]` for the title
+    - Use a clear and descriptive title
+- **Body** - include the following sections:
+    - System environment (OS, iroha version)
+    - Steps to reproduce
+    - Expected behavior
+    - Actual behavior
+
+
+
+### Suggesting Enhancements
+
+An *enhancement* is a code or idea, which makes **existing** code or design faster, more stable, portable, secure or better in any other way.
+
+Enhancements are tracked as [GitHub Issues](https://guides.github.com/features/issues/). To submit new enhancement, create new Issue and incllude these details:
+
+- **Title**
+    - Write prefix `[Enhancement]`
+    - Use a clear and descriptive title
+- **Body** - include the following sections:
+    - *Target* - what is going to be improved?
+    - *Motivation* - why do we need it?
+    - *Description* - how to implement it?
+
+
+
+### Asking Questions
+
+A *question* is any discussion that is typically neigher a bug, nor feature request, nor improvement - "How do I do X?".
+
+Questions are tracked as [Github Issues](https://guides.github.com/features/issues/) or via private messages in [your favourite messenger](#contact-developers).
+
+To submit new question in GitHub Issues, it must include these details:
+
+- **Title**
+    - Write prefix `[Question]`
+    - Use a clear and descriptive title
+- **Body** - describe your question with as many details as possible.
+
+
+
+### Your First Code Contribution
+
+Read our [C++ Style Guide](#c++-style-guide) and start with beginner-friendly issues with label [`[good-first-issue]`](https://github.com/hyperledger/iroha/issues?q=is:open+is:issue+label:good-first-issue ). Indicate somehow that you are working on this task.
+
+
+
+### Pull Requests
+
+- Fill in [the required template](.github/PULL_REQUEST_TEMPLATE.md)
+
+- **Write tests** for new code.
```

Comment:
can you mention coverage? like 70%

Reply:
May be just *reasonable* is enough?
For me it's better accept nice contributions with some meaningful tests that doesn't cover small and rare use-cases.